evm/utils/state_access_restriction.py

```diff
@@ -0,0 +1,36 @@
+from evm.constants import (
+    NONCE_TRIE_PREFIX,
+    BALANCE_TRIE_PREFIX,
+    CODE_TRIE_PREFIX,
+    STORAGE_TRIE_PREFIX,
+)
+
+from evm.utils.keccak import (
+    keccak,
+)
+
+
+def is_accessible(address, slot_as_key, access_prefix_list):
+    """Check if a storage slot is specified in an access prefix list."""
+    key = keccak(address) + STORAGE_TRIE_PREFIX + slot_as_key
+    for prefix in access_prefix_list:
+        if key.startswith(prefix):
+            return True
+    return False
+
+
+def to_prefix_list_form(access_list):
+    """Expand an access list to a flat list of storage key prefixes.
+
+    As input a list of the form `[[address, prefix1, prefix2, ...], ...]` is expected.
+    """
+    prefix_list = []
```

Comment:
Most of py-evm avoids this pattern of creating a list, filling it, then returning it using the `eth_utils.to_tuple` decorator.

```python
def build_powers_with_mutation():
    the_values = []
    for i in range(10):
        the_values.append(2**i)
    return the_values

@to_tuple
def build_powers_no_mutation():
    for i in range(10):
        yield 2**i
```